tests/e2e/helpers/decompressors/base-decompressors-populator.ts

```diff
@@ -0,0 +1,69 @@
+import { createClient } from 'redis';
+
+export abstract class BaseDatabasePopulator {
+    private client: ReturnType<typeof createClient>;
+
+    protected abstract createCompressedKeys(): Promise<void>;
+
+    constructor(private host: string, private port: string) {
+        const dbConf = { port: Number.parseInt(port), host, username: 'default' };
+        this.client = createClient(dbConf);
+
+        this.client.on('error', (error: string) => {
+            throw new Error(`Redis connection error: ${error}`);
+        });
+    }
+
+    public async populateDB(): Promise<void> {
+        this.client.on('connect', async () => {
+            console.log('Connected to Redis');
+            try {
+                await this.createCompressedKeys();
+            } catch (error) {
+                console.error('Error during key creation:', error);
+            } finally {
+                await this.client.quit();
+            }
+        });
+    }
+
+    protected async createHash(
```

Comment:
better to pass the number of fields as an argument instead of onlyOneItem
and probably you can avoid using value2

Reply:
fixed